packages/core/src/client.ts

```diff
@@ -2692,6 +2697,16 @@ export class MedplumClient extends EventTarget {
       // Silently ignore if this environment does not support storage events
     }
   }
+
+  private retryCatch(retryNumber: number, maxRetries: number, err: Error): void {
+    // This is for the 1st retry to avoid multiple notifications
+    if (retryNumber === 1) {
+      this.dispatchEvent({ type: 'offline' });
+    }
+    if (retryNumber === maxRetries - 1) {
```

Comment:
possibly update logic to check `retryNumber >= maxRetries - 1` and a comment that the exception only thrown on the last retry attempt.